src/modules/cmdpal/Microsoft.CmdPal.UI/Helpers/OptionalColorToBrushConverter.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CmdPal.Extensions;
+using Microsoft.UI;
+using Microsoft.UI.Xaml.Media;
+using Color = Windows.UI.Color;
+
+namespace Microsoft.CmdPal.UI.Helpers;
+
+public static partial class OptionalColorBrushCacheProvider
+{
+    private static readonly Dictionary<OptionalColor, SolidColorBrush> _brushCache = [];
+
+    //// TODO: Not sure how best to make this configurable or provide a themed resource here... (mostly due to shortcomings of x:Bind/MarkupExtension, need to file many bugs)
+    private static readonly SolidColorBrush _defaultBackgroundBrush = new(Colors.Black);
+    private static readonly SolidColorBrush _defaultForegroundBrush = new(Colors.White);
+
+    public static SolidColorBrush? Convert(OptionalColor color, bool isForeground)
+    {
+        if (!color.HasValue)
```

Comment:
I almost think we want to pass in both the foreground and background OptionalColors here, cause:

* If there's a foreground but no background, we want the text and border to be the FG color
* if there's no FG or BG, we want the text to be default colored, and the border to be hidden (so it's just some text)
* if there's a background but no foreground, I think it's okay to have the text be "transparent" / punch out
  * You know, as soon as  typed that up, I remember that's incredibly difficult to actually do. 

okay you know what, maybe that's harder than I thought

Reply:
@niels9001 want to think about these scenarios a bit? I can sync with you this week on what other types of adjustments we'd want to configure on this and then how we can tackle it from a converter/function/helper point of view (as this ties into all the same issues we've been facing with the discussion on the color ramp helpers in the Toolkit as well).